src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -672,71 +678,82 @@ const static auto formats = {
     std::locale(std::locale::classic(),
                 new boost::local_time::local_time_input_facet("%Y-%m-%d %H:%M:%s%ZP")),
 };
-const static boost::posix_time::ptime epoch(boost::gregorian::date(1970, 1, 1));
-const static boost::posix_time::ptime max_date(boost::gregorian::date(2030, 1, 1));
 
-static std::chrono::milliseconds parseDateAsMillis(
-    const std::string& datetime, const boost::posix_time::ptime& default_time = epoch) {
-    boost::posix_time::ptime utc_time{boost::posix_time::not_a_date_time};
-    auto millis = 0LL;
+class DateToIntGenerator : public Generator<int64_t> {
+public:
+    explicit DateToIntGenerator(UniqueGenerator<bsoncxx::types::b_date>& generator) {
+        std::swap(_generator, generator);
```

Comment:
Is this the correct approach here?

Reply:
I'm not sure what the intent here is and why the `std::swap` does what you want? I don't usually see `swap` used with refs - usually it's with value types or values. This could be totally kosher but I for one could use a comment here about why this is the right thing.